onnxruntime/core/providers/cuda/shared_inc/fpgeneric.h

```diff
@@ -174,6 +174,270 @@ inline cublasStatus_t cublasGemmHelper(cublasHandle_t, cublasOperation_t, cublas
+                                           cublasOperation_t transa,
+                                           cublasOperation_t transb,
+                                           int m, int n, int k,
+                                           const float* alpha,
+                                           const float* A, int lda,
+                                           const float* B, int ldb,
+                                           const float* beta,
+                                           float* C, int ldc,
+                                           cudaStream_t stream) {
+  cudaDataType_t data_type = CUDA_R_32F;
+  cudaDataType_t scale_type = CUDA_R_32F;
+  cublasComputeType_t compute_type = CUBLAS_COMPUTE_32F;
+
+  cublasLtMatrixLayout_t A_desc = NULL, B_desc = NULL, C_desc = NULL;
+  cublasLtMatmulDesc_t operation_desc = NULL;
+
+  auto clean_desc_A = gsl::finally([&A_desc]() {
+    if (A_desc) {
+      cublasLtMatrixLayoutDestroy(A_desc);
+    }
+  });
+
+  auto clean_desc_B = gsl::finally([&B_desc]() {
+    if (B_desc) {
+      cublasLtMatrixLayoutDestroy(B_desc);
+    }
+  });
+
+  auto clean_desc_C = gsl::finally([&C_desc]() {
+    if (C_desc) {
+      cublasLtMatrixLayoutDestroy(C_desc);
+    }
+  });
+
+  auto clean_matmul_desc = gsl::finally([&operation_desc]() {
+    if (operation_desc) {
+      cublasLtMatmulDescDestroy(operation_desc);
+    }
+  });
+
+  if (Status::OK() != InitializeCublasLtMatmulDescAndOperationHelper(A_desc, lda,
+                                                                     transa,
+                                                                     B_desc, ldb,
+                                                                     transb,
+                                                                     C_desc, ldc,
+                                                                     data_type,
+                                                                     m, n, k,
+                                                                     operation_desc,
+                                                                     compute_type,
+                                                                     scale_type)) {
+    return CUBLAS_STATUS_ALLOC_FAILED;
+  }
+
+  // TODO (hasesh): Allow CublasLtMatmul tuning for clients by allowing them to pass in the workspace and algo of their choice.
+  // According to the cublasLtMatmul documentation, passing in NULL for the algo means that "an implicit heuristics query with
+  // default search preferences will be performed to determine actual algorithm to use". Source: cublasLtMatmul documentation.
+  return cublasLtMatmul(
+      handle, operation_desc, reinterpret_cast<const void*>(alpha),
+      A, A_desc, B, B_desc,
+      reinterpret_cast<const void*>(beta),
+      C, C_desc,
+      C, C_desc,
+      /*algo*/ NULL, /*workspace memory*/ NULL, /*workspace size*/ 0, stream);
```

Comment:
By default, we pass NULL for the algo to employ Cublas heuristics to pick the Matmul algo.  We also can pass in DEFAULT_ALGO and explicitly ask Cublas to use that without employing heuristics. (We can pick he best approach based on benchmarking results).

In future, we can support automated Gemm algo tuning/picking but that will be in a different PR.